common/membership/ringpop/factory.go

```diff
@@ -144,13 +145,25 @@ func (factory *factory) getMonitor() *monitor {
 			factory.broadcastAddressResolver,
 			factory.Config.MaxJoinDuration,
 			maxPropagationTime,
-			time.Time{}, // TODO: set this based on dynamic config
+			factory.getJoinTime(maxPropagationTime),
 		)
 	})
 
 	return factory.monitor
 }
 
+func (factory *factory) getJoinTime(maxPropagationTime time.Duration) time.Time {
+	var alignTime time.Duration
+	switch factory.ServiceName {
+	case primitives.MatchingService:
+		alignTime = factory.DC.GetDurationProperty(dynamicconfig.MatchingAlignMembershipChange, 0*time.Second)()
+	}
+	if alignTime == 0 {
+		return time.Time{}
```

Comment:
Do we need to return a valid time here?

Reply:
Any time in the past (which the zero time is) will disable the delayed join mechanism